src/utils/PayloadValidation.ts

```diff
@@ -16,14 +18,25 @@ const checkIfTankOrBattery = (payload: ITechRecord) => {
   return isTankOrBattery;
 };
 
-export const validatePayload = (payload: ITechRecord) => {
+const featureFlagValidation = (validationSchema: ObjectSchema, payload: ITechRecord, validateEntireRecord: boolean, context: any) => {
+  const allowAdrUpdatesOnlyFlag: boolean = Configuration.getInstance().getAllowAdrUpdatesOnlyFlag();
+  if (allowAdrUpdatesOnlyFlag && !validateEntireRecord) {
+    Object.assign(context, {stripUnknown: true});
+    return validateOnlyAdr.validate({adrDetails: payload.adrDetails, reasonForCreation: payload.reasonForCreation}, context);
```

Comment:
could use destructuring here to make it more concise
const { adrDetails, reasonForCreation } = payload

return validateOnlyAdr.validate({adrDetails, reasonForCreation}, context)